src/commissaire_http/router/__init__.py

```diff
@@ -30,6 +30,42 @@ class Router(Mapper):
     #: Class level logger
     logger = logging.getLogger('Router')
 
+    def __init__(self, optional_slash=False, *args, **kwargs):
+        """
+        :param optional_slash: If /'s are optional when matching directories.
+        :type optional_slash: bool
+        :param args: All non-keyword arguments.
+        :type args: tuple
+        :param kwargs: All other keyword arguments.
+        :type kwargs: dict
+        """
+        super().__init__(*args, **kwargs)
+        self._optional_slash = optional_slash
+
+    def connect(self, *args, **kwargs):
+        """
+        Overrides Mapper.connect adding in support for optional slashses.
+
+        :param args: All non-keyword arguments.
+        :type args: tuple
+        :param kwargs: All other keyword arguments.
+        :type kwargs: dict
+        """
+        # Cast to a list so we can modify the args
+        args = list(args)
+        # If we are asked to use an optional slash then find the url_path
+        # in the call. It is either the first or second element.
+        if self._optional_slash:
+            url_path_idx = 0
+            if len(args) > 1:
+                url_path_idx = 1
+            # If the url path ends with a forward slash then append the
+            # paramter regex to for matching and store it in _.
```

Comment:
typo: `paramter` -> `parameter`

Also, "to for matching" sounds odd but not sure what the proper correction is.  "for matching"?